File: app/src/main/java/com/vktiwari/pagingretrofitrxjava/datasource/PeopleDataSourceFactory.java
package com.vktiwari.pagingretrofitrxjava.datasource;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;

import com.vktiwari.pagingretrofitrxjava.model.People;

public class PeopleDataSourceFactory extends DataSource.Factory<Integer, People> {
    private MutableLiveData<PeopleDataSource> mutableLiveData;

    public PeopleDataSourceFactory() {
        mutableLiveData = new MutableLiveData<>();
    }

    @Override
    public DataSource<Integer, People> create() {
        PeopleDataSource peopleDataSource = new PeopleDataSource();
        mutableLiveData.postValue(peopleDataSource);
        return peopleDataSource;
    }

    public MutableLiveData<PeopleDataSource> getMutableLiveData() {
        return mutableLiveData;
    }
}

File: app/src/kotlin/kotlin/com/vktiwari/pagingretrofitrxjava/datasource/PeopleDataSourceFactory.kt
package com.vktiwari.pagingretrofitrxjava.datasource

import android.arch.lifecycle.MutableLiveData
import android.arch.paging.DataSource
import com.vktiwari.pagingretrofitrxjava.model.People

class PeopleDataSourceFactory : DataSource.Factory<Int,People>(){
    private var mutableLiveData: MutableLiveData<PeopleDataSource> = MutableLiveData()

    override fun create(): DataSource<Int, People> {
        val peopleDataSource = PeopleDataSource()
        mutableLiveData.postValue(peopleDataSource)
        return peopleDataSource
    }

    fun getMutableLiveData(): MutableLiveData<PeopleDataSource> {
        return mutableLiveData
    }
}